include/boost/math/special_functions/fast_float_distance.hpp

```diff
@@ -0,0 +1,132 @@
+//  (C) Copyright Matt Borland 2022.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_SF_FAST_FLOAT_DISTANCE
+#define BOOST_MATH_SF_FAST_FLOAT_DISTANCE
+
+#include <boost/math/special_functions/next.hpp>
+#include <boost/math/tools/throw_exception.hpp>
+#include <stdexcept>
+#include <limits>
+
+#if defined(BOOST_MATH_USE_FLOAT128) && !defined(BOOST_MATH_STANDALONE)
+#include <boost/multiprecision/float128.hpp>
+#include <boost/multiprecision/detail/standalone_config.hpp>
+#define BOOST_MATH_USE_FAST_FLOAT128
+#elif defined(BOOST_MATH_USE_FLOAT128) && defined(BOOST_MATH_STANDALONE)
+#  if __has_include(<quadmath.h>)
+#    include <quadmath.h>
+#    define BOOST_MATH_USE_FAST_STANDALONE_FLOAT128
+#  endif
+#endif
+
+namespace boost { namespace math { 
+
+// https://randomascii.wordpress.com/2012/01/23/stupid-float-tricks-2/
+// https://blog.regehr.org/archives/959
+inline std::int32_t fast_float_distance(float a, float b)
```

Comment:
I think float distance can overflow an int32 from two floats right?

Reply:
Yes. For all the edge cases I defer to John's implementation which is tested [here](https://github.com/boostorg/math/blob/develop/test/test_next.cpp#L192) to avoid this issue.

Reply:
Well, I might just let the return type be a `std::int64_t`; seems like very little loss of performance.

Reply:
I think that will emit a widening warning since we have to `memcpy` `float` into `std::int32_t`.

Reply:
How about something like

```
int32_t i,j;
memcpy(&j, &a);
memcpy(&i, &b);
return int64_t(i) - int64_t(j);
```

Reply:
That would work for `float`, but the logic is shared with the `double`. The same issue would occur, and I don't want to use `__int128_t` in that case.

Reply:
Yes, you care correct.